src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchValidationSingleSwFlowSpec.groovy

```diff
@@ -285,12 +295,7 @@ class SwitchValidationSingleSwFlowSpec extends HealthCheckSpecification {
 
         cleanup:
         flow && !deleteFlow && flowHelperV2.deleteFlow(flow.flowId)
-        if (!testIsCompleted) {
-            northbound.synchronizeSwitch(sw.dpId, true)
-            Wrappers.wait(RULES_INSTALLATION_TIME) {
-                assert northbound.getSwitchRules(sw.dpId).flowEntries*.cookie.sort() == sw.defaultCookies.sort()
-            }
-        }
+        flow && !testIsCompleted && synchronizeAndValidateRules(sw)
```

Comment:
You don't need flow && here, I guess.

Reply:
There is no need to sync the switch when no flow has been added.